src/utils/gitUtils.ts

```diff
@@ -21,10 +21,25 @@ import { getSingleRootFsPath } from './workspaceUtils';
 export type GitWorkspaceState = { repo: Repository | null, dirty: boolean, remoteRepo: ReposGetResponseData | undefined; hasAdminAccess: boolean };
 export type VerifiedGitWorkspaceState = GitWorkspaceState & { repo: Repository };
 
+// use only when using the gitApi
+export async function callWithGitErrorHandling<T>(callback: () => Promise<T>): Promise<T | null> {
+    try {
+        return await callback();
+    } catch (gitError) {
+        // ignore empty commit errors which will happen if a user initializes a blank folder or have changes in a nested git repo
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```

Comment:
Can we "fix" this error instead of disabling it? Ideally use `parseError`, but at least make sure it's an object and more gracefully handle the types. Would be embarrassing if our error handling logic had it's own error

Reply:
I'm not entirely sure what you mean by fixing it.  Do you mean fix the cause of this error in our extension? 

I've tried to mitigate it by preventing users from creating SWAs if their workspace is completely empty.  The nested repository case seems unlikely, but I ran into it and was having issues with VS Code reporting that I had git changes, but when I actually tried to commit, it was throwing an error saying that there were no changes.  I actually haven't been able to reproduce it anymore so maybe they're working on it.  There is an issue following nested repos: https://github.com/microsoft/vscode/issues/37947

`parseError` doesn't work in this case because it just sets the message equal to `Failed to execute git` since that's what all the gitApi commands throw up.  All the git terminal information is in the `stderr` and `stdout`.